src/main/scala/is/hail/expr/ir/functions/MathFunctions.scala

```diff
@@ -79,5 +80,21 @@ object MathFunctions extends RegistryFunctions {
     registerJavaStaticFunction("%", TInt64(), TInt32(), TInt64())(jMathClass, "floorMod")
 
     registerScalaFunction("isnan", TFloat64(), TBoolean())(thisClass, "isnan")
+
+    registerCodeWithMissingness("&&", TBoolean(), TBoolean(), TBoolean()) { (_, l, r) =>
+      EmitTriplet(
+        Code(l.setup, r.setup),
+        l.m || (coerce[Boolean](l.v) && r.m),
+        coerce[Boolean](l.v) && coerce[Boolean](r.v)
```

Comment:
does the method builder provide variable binding? we should avoid evaluating `l.v` twice if possible.